testng-core/src/test/kotlin/test/SimpleBaseTest.kt

```diff
@@ -0,0 +1,489 @@
+package test
+
+import org.assertj.core.api.Assertions
+import org.testng.*
+import org.testng.annotations.ITestAnnotation
+import org.testng.collections.Lists
+import org.testng.internal.annotations.AnnotationHelper
+import org.testng.internal.annotations.DefaultAnnotationTransformer
+import org.testng.internal.annotations.IAnnotationFinder
+import org.testng.internal.annotations.JDK15AnnotationFinder
+import org.testng.xml.*
+import org.testng.xml.internal.Parser
+import java.io.*
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.*
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+import kotlin.collections.ArrayList
+
+open class SimpleBaseTest {
+
+    companion object Companion {
+        @JvmStatic
+        val TEST_RESOURCES_DIR = "test.resources.dir"
+
+        @JvmStatic
+        fun run(vararg testClasses: Class<*>) : InvokedMethodNameListener {
+            return run(false, *testClasses)
+        }
+
+        @JvmStatic
```

Comment:
```kotlin
@JvmStatic
        private fun run(skipConfiguration: Boolean = false, tng: TestNG) = InvokedMethodNameListener(skipConfiguration).let {
            tng.addListener(listener)
            tng.run()
        }
```

Reply:
can be replace by something like:

```kotlin
fun TestNG.run(skipConfiguration: Boolean = false) = InvokedMethodNameListener(skipConfiguration).let {
            this.addListener(it)
            this.run()
        }
```

Reply:
How would this replace the existing method need that all the other methods in this base class have ?

Reply:
Refactored